model/src/main/kotlin/ResolvedConfiguration.kt

```diff
@@ -53,8 +54,8 @@ data class ResolvedConfiguration(
     /**
      * All resolutions that apply to this [OrtResult].
      */
-    @JsonInclude(JsonInclude.Include.NON_NULL)
-    val resolutions: Resolutions? = null
+    @JsonInclude(JsonInclude.Include.CUSTOM, valueFilter = ResolutionsFilter::class)
+    val resolutions: Resolutions = Resolutions()
```

Comment:
I don't remember anymore why we're using a custom `valueFilter` in these cases instead of just `JsonInclude.Include.NON_DEFAULT`. Can you remind me? 1fec27e3dd6789c21f11b3d273f08a890a6315f5 does not really give a clue.

Reply:
The property is nullable on purpose. If it is `null` it means no resolutions were resolved so far. An empty `Resolutions` object means that resolutions were resolved but none were found. I should have added this to the property docs.
Removing nullability breaks the handling in the  `ReporterCommand`.

Reply:
Hm, I've adjusted the handling in the `ReporterCommand`. It seemed to me it does not need to be able to distinguish between these two cases. Have you noticed the changes there?

Reply:
I've split out some hopefully uncontroversial bits to https://github.com/oss-review-toolkit/ort/pull/6745

Reply:
sure.